---
title: "MDS plot & Heatmap"
author: "Evangelyn Sim"
date: "2/2/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




## Introduction

In the GEO submission 3 processed files were uploaded. 

1. hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix_filt.csv
2. hrna_dev_mf_fulllen_se_strrev_q30.mx.all.MvsF.fix_filt.csv
3. hrna_dev_mf_fulllen_se_strrev_q30.mx.all_unfiltered.csv

They have been uploaded in the /output folder and will be used below to generate different figures.



### Used libraries and functions

```{r}
library(edgeR)
library(limma)
library(Glimma)
library(gplots)

library(ComplexHeatmap)
library(circlize)
library(RColorBrewer)
library(mclust)

```


### Read files 
```{r}

rm1 <- read.csv("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Human_Development_RNAseq_bulk/output/hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix_filt.csv", row.names = 1)

info = read.delim("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Human_Development_RNAseq_bulk/output/RNAseq_samplesheet.txt", header = TRUE, sep = "\t")

m = match(info$ID,names(rm1))
rm2 = rm1[,m]

rm1 = rm2

sampleinfo = info
levels(factor(sampleinfo$Group))
sampleinfo$colour = c("violetred4","darkgoldenrod2","tomato")[factor(sampleinfo$Group)]
table(colnames(rm2)==sampleinfo$ID)

y <- DGEList(rm2)

```

# Make plot

```{r, fig.height=4,fig.width=10}

par(mfrow=c(1,3))
#par(mar=c(5,1,5,1))

plotMDS(y, pch=c(0,1,2)[factor(sampleinfo$Group)], col=sampleinfo$colour, cex = 2)
legend("topleft", legend = c("Fetal","Young","Adult"), pch=c(1,2,0), col = c("darkgoldenrod2","tomato","violetred4"), cex= 1.5)

plotMDS(y, pch=c(0,15,1,16,2,17)[factor(sampleinfo$BinSex)], col=sampleinfo$colour, cex = 2)
legend("topleft", legend = c("Fetal","Young","Adult"), pch=c(1,2,0), col = c("darkgoldenrod2","tomato","violetred4"), cex= 1.5)
legend("top", legend = c("F","M"), pch=c(1,16), col = c("grey"), cex=1.5)

plotMDS(y, cex = 0.8)


```


# Remove Chr X & Y genes

```{r}

rm3 = read.delim("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Human_Development_RNAseq_bulk/output/hrna_dev_mf_fulllen_se_strrev_q30.mx.chr", header = TRUE, row.names = 1)
rm3 = merge(rm3, rm2, by = "row.names")
rm3 = rm3[!grepl("Y",rm3$Chr),]
rm3 = rm3[!grepl("X",rm3$Chr),]
row.names(rm3)= rm3$Row.names
rm3 = rm3[,c(7:ncol(rm3))]

z <- DGEList(rm3)

```

# Make MDS plot (removed Chr X & Y genes)

```{r, fig.height=4,fig.width=10}

par(mfrow=c(1,3))

plotMDS(z, pch=c(0,1,2)[factor(sampleinfo$Group)], col=sampleinfo$colour, cex = 2)
legend("topleft", legend = c("Fetal","Young","Adult"), pch=c(1,2,0), col = c("darkgoldenrod2","tomato","violetred4"), cex= 1.5)

plotMDS(z, pch=c(0,15,1,16,2,17)[factor(sampleinfo$BinSex)], col=sampleinfo$colour, cex = 2)
legend("topleft", legend = c("Fetal","Young","Adult"), pch=c(1,2,0), col = c("darkgoldenrod2","tomato","violetred4"), cex= 1.5)
legend("top", legend = c("F","M"), pch=c(1,16), col = c("grey"), cex=1.5)

plotMDS(z, cex = 0.8)


```


# Make Heatmap (removed Chr X & Y genes)

```{r}
mycpm = cpm(rm3, log = T)

corr1 = cor(mycpm, method = "pearson")

matrix = as.matrix(corr1)

set.seed(20)

mypalette <- brewer.pal(11,"RdYlBu")
morecols <- colorRampPalette(mypalette) #colorramppa make colour gradient

ann <- data.frame(sampleinfo[,c(1,4)])
colnames(ann) <- c("Group","Sex")
colours <- list("Group"=c("Fetal"="darkgoldenrod2","Young"="tomato","Adult"="violetred4" ),
                "Sex"=c("F"="deeppink1","M"="dodgerblue1"))
colAnn <- HeatmapAnnotation(df=ann, which="col", col=colours, annotation_width=unit(c(1, 4), "cm"), gap=unit(1, "mm"))
colAnn1 <- HeatmapAnnotation(df=ann, which="row", col=colours, annotation_width=unit(c(1, 4), "cm"), gap=unit(1, "mm"))

```

```{r}

ht_list = 
  Heatmap(matrix, name = "Correlation",
          row_title = "",
          row_title_gp = gpar(fontsize = 8),
          col = rev(morecols(50)),
          width = unit(7, "cm"),
          height = unit(7, "cm"),
          top_annotation = colAnn,
          left_annotation = colAnn1,
          cluster_rows =T,
          show_row_names = T,
          row_names_side = "left",
          row_names_gp = gpar(fontsize = 8),
          cluster_columns =T,
          column_names_side = "top",
          column_names_gp = gpar(fontsize = 8),  
          column_title = "", 
          column_title_gp = gpar(fontsize = 8))

set.seed(20)
clus = draw(ht_list)

```

